src/main/ts/components/Editor.tsx

```diff
@@ -310,194 +230,310 @@ export class Editor extends React.Component<IAllProps> {
+  const bindHandlers = React.useCallback(() => {
+    if (editorRef.current !== undefined) {
+      // typescript chokes trying to understand the type of the lookup function
+      configHandlers(editorRef.current, props, boundHandlersRef.current, (key) => props[key] as any);
+      // check if we should monitor editor changes
+    }
+  }, [ props ]);
+
+  const handleBeforeInput = React.useCallback((_evt: EditorEvent<unknown>) => {
+    const content = editorRef.current?.getContent();
+    if (value !== undefined && value === content && editorRef.current) {
+      if (!inline || editorRef.current.hasFocus()) {
         try {
           // getBookmark throws exceptions when the editor has not been focused
           // possibly only in inline mode but I'm not taking chances
-          this.valueCursor = this.editor.selection.getBookmark(3);
+          setValueCursor(editorRef.current.selection.getBookmark(3));
         } catch (e) { /* ignore */ }
       }
     }
-  };
+  }, [ value, inline ]);
 
-  private handleBeforeInputSpecial = (evt: EditorEvent<KeyboardEvent>) => {
+  const handleBeforeInputSpecial = React.useCallback((evt: EditorEvent<KeyboardEvent>) => {
     if (evt.key === 'Enter' || evt.key === 'Backspace' || evt.key === 'Delete') {
-      this.handleBeforeInput(evt);
+      handleBeforeInput(evt);
     }
-  };
+  }, [ handleBeforeInput ]);
 
-  private handleEditorChange = (_evt: EditorEvent<unknown>) => {
-    const editor = this.editor;
-    if (editor && editor.initialized) {
-      const newContent = editor.getContent();
-      if (this.props.value !== undefined && this.props.value !== newContent && this.props.rollback !== false) {
+  const handleEditorChange = React.useCallback((_evt: EditorEvent<unknown>) => {
+    const currentEditor = editorRef.current;
+    if (currentEditor && currentEditor.initialized) {
+      const newContent = currentEditor.getContent();
+      if (value !== undefined && value !== newContent && rollback !== false) {
         // start a timer and revert to the value if not applied in time
-        if (!this.rollbackTimer) {
-          this.rollbackTimer = window.setTimeout(
-            this.rollbackChange,
-            typeof this.props.rollback === 'number' ? this.props.rollback : 200
+        if (!rollbackTimerRef.current) {
+          rollbackTimerRef.current = window.setTimeout(
+            rollbackChange,
+            typeof rollback === 'number' ? rollback : 200
           );
         }
       }
 
-      if (newContent !== this.currentContent) {
-        this.currentContent = newContent;
-        if (isFunction(this.props.onEditorChange)) {
-          this.props.onEditorChange(newContent, editor);
+      if (newContent !== currentContentRef.current) {
+        currentContentRef.current = newContent;
+        if (isFunction(onEditorChange)) {
+          onEditorChange(newContent, currentEditor);
         }
       }
     }
-  };
+  }, [ value, onEditorChange, rollback, rollbackChange ]);
```

Comment:
There is no way you want this to regenerate on changing `value`... The only way this will work is if it references `value` though a ref. I suspect you'll have to reference everything through refs because this is a TinyMCE event handler and unless you are constantly deregistering the old version and re-registering the new version this is not going to work.

Reply:
(yes it will be a kind of hell getting this to work - that is why I never moved off the class based implementation)